wandb/sdk/launch/agent/agent.py

```diff
@@ -194,7 +200,11 @@ def run_job(self, job: Dict[str, Any]) -> None:
         run = backend.run(project, builder, registry_config)
         if run:
             self._jobs[run.id] = run
-            self._running += 1
+            if launch_spec.get("uri") == "placeholder-uri-scheduler":
```

Comment:
Non-blocking but we might want to consider either
1. Making this string a reusable constant since it seems super important that this is identical to [these other usages](https://github.com/search?q=org%3Awandb+placeholder-uri-scheduler&type=code) and it would be nice to have it defined in one place; or
2. Rethinking how we identify that a job is a sweep scheduler job, e.g. having a field in the launch spec that declares if this is a special purpose job, and what the special purpose is (`sweep_scheduler` in this case, but could be reused for other purposes in the future). Hijacking the "uri" field feels hacky to me. This is almost definitely outside the scope of this change, though.

Reply:
Yeah great comment. Definitely a hold-over from Hugo's implementation that isn't great. I'll make sure to think about this when returning to Optuna --> refactoring some things. For now I'll make that constant thanks :)